pkg/rest/rest.go

```diff
@@ -0,0 +1,141 @@
+package rest
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/crc-org/vfkit/pkg/rest/define"
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+)
+
+// VFKitService is used for the restful service; it describes
+// the variables of the service like host/path but also has
+// the router object
+type VFKitService struct {
+	Host   string
+	Path   string
+	Port   int
+	router *mux.Router
+	Scheme ServiceScheme
+}
+
+// Start initiates the already configured restful service
+func (v *VFKitService) Start() error {
+	logrus.Debugf("starting rest service on %s", v.Host)
+	return http.ListenAndServe(v.Host, v.router)
+}
+
+// NewServer creates a new restful service
+func NewServer(vm *VzVirtualMachine, scheme ServiceScheme, uri *url.URL) *VFKitService {
+	r := mux.NewRouter()
+	s := VFKitService{
+		router: r,
+		Host:   uri.Host,
+		Scheme: scheme,
+	}
+	// Handlers for the restful service.  This is where endpoints are defined.
+	r.HandleFunc("/vm/state", vm.getVMState()).Methods(http.MethodGet)
+	r.HandleFunc("/vm/state", vm.setVMState()).Methods(http.MethodPost)
+	r.HandleFunc("/vm/inspect", vm.inspect()).Methods(http.MethodGet)
```

Comment:
I don't expect vfkit and vz code to be fully thread safe, I don't know if the handlers are called in their own go routines, or if gorilla/its replacement run all of them in the same go routine, ...